compiler/erg_common/traits.rs

```diff
@@ -347,37 +347,38 @@ pub trait LimitedDisplay {
     fn limited_fmt(&self, f: &mut std::fmt::Formatter<'_>, limit: usize) -> std::fmt::Result;
 }
 
-// for Runnable::run
-fn expect_block(src: &str) -> bool {
-    let src = src.trim_end();
-    src.ends_with(&['.', '=', ':'])
-        || src.ends_with("->")
-        || src.ends_with("=>")
-        // when `"""` are on the same line
-        // e.g. """something"""
-        || src.contains("\"\"\"") && src.find("\"\"\"") == src.rfind("\"\"\"")
+#[derive(Debug, PartialEq, Eq)]
+pub enum BlockKind {
+    Main,          // now_block Vec must contain this
+    Assignment,    // =
+    Lambda,        // =>, ->, do:, do!:
```

Comment:
I have included the `:colon` in the Lambda
Is there a better way to handle this?

Reply:
you mean like this?

```python
match! x:
    ...
```

Defining an additional enum option with a name like `ColonCall` may be better.

Reply:
Thank you